docs/source/conf.py

```diff
@@ -27,11 +27,28 @@
 from sphinx_gallery.sorting import ExplicitOrder
 
 TMP_FOLDER = Path(__file__).parents[2] / "tmp"
+TMP_FOLDER.mkdir(exist_ok=True)
+
+# Generate a JSON compatibility table
+
+html_extra_path = []
+
+with open("additional/releases.yaml") as f:
+    compat_table = yaml.safe_load(f)
+    dest = Path(TMP_FOLDER, "releases.json")
+    with open(dest, "w") as f:
+        json.dump(compat_table, f)
+        html_extra_path.append(str(dest))
+
+repo = git.Repo(search_parent_directories=True)
+current_commit = repo.head.commit.hexsha
+tags_commit = [tag.commit.hexsha for tag in repo.tags]
 
-if os.environ.get("READTHEDOCS_VERSION_TYPE") == "tag":
-    SUBSTRA_VERSION = "0.43.0"
-    TOOLS_VERSION = "0.20.0"
-    SUBSTRAFL_VERSION = "0.35.1"
+if os.environ.get("READTHEDOCS_VERSION_TYPE") == "tag" or current_commit in tags_commit:
```

Comment:
Maybe you can rely on the release table there too: `if repo.commit(compat_table["releases"][0]["version"]) == repo.head.commit`. But it's fine as-is IMO, it could break if we use tags for stuff other than releases but we don't.

Reply:
Yep that was my concern... One way would be to check the name of the tags to match the semver format. But with the release candidate and all it's not that obvious...